### DISCUSSION TITLE
case sensitivity and avoiding tag name duplication

### DISCUSSION THREAD

**Discussion by devcbash** [spatie/laravel-tags#501] [Q&A]
hi,
 I am using your package but am interested to know how you deal with tag name duplication?
 I am having an issue with how to manage the case sensitive inputs of tag names. for example tags like: "summer","Summer", I required only one unique tag name  for later filtering and presentation. 

In addition any drawbacks for changing the column type to varchar(255) in mysql as it adds caase sensitivity as opposed to the json column type?
 In addition would you save the tags as lower case or capitalized? thanks

**Comment by JannesDM-VMA** [comment]
I don't know if this helps; but we are using this package, and we save all tags in lowercase (but it's personal preference I think).
A part of the duplication is handled by using the `findOrCreate` method for finding **and** creating tags with this package.
I'm sure it's possible to add your own code such that every tag you create/update is being cast to lowercase first.
Maybe this can help: [read-the-docs!](https://spatie.be/docs/laravel-tags/v3/advanced-usage/using-your-own-tag-model)

Be mindful that the docs are for V3, you can find the changes here: [changelog](https://github.com/spatie/laravel-tags/blob/main/CHANGELOG.md).
Happy coding!
